PC/launcher.c

```diff
@@ -1600,14 +1601,15 @@ process(int argc, wchar_t ** argv)
     }
     else {
         p = argv[1];
-        plen = wcslen(p);
         if ((argc == 2) && 
-            (!wcsncmp(p, L"-0", wcslen(L"-0")) || /* Starts with -0 or --list */
-            !wcsncmp(p, L"--list", wcslen(L"--list"))))
+            (!_wcsicmp(p, L"-0") || !_wcsicmp(p, L"--list") || /* Version args. */
+            !_wcsicmp(p, L"-0p") || !_wcsicmp(p, L"--list-paths")))
         {
-            valid = show_python_list(argv); /* Check for -0 or --list FIRST */
+            show_python_list(argv);
+            version_call = TRUE;
+            executable = NULL;
```

Comment:
Why not just return immediately from here? There doesn't seem to be any cleanup that has to happen.

Reply:
I wrote it to imitate the behavior of -h where it'll search for any Python version and return `RC_NO_PYTHON` if nothing is found. If that's not desirable, could just bump the check up a level.